_includes/coinduction-extreme-predicates-ordinals.html

```diff
@@ -0,0 +1,1744 @@
+</p>
+<pre class="para-block pre-fenced pre-fenced3 language-dafnyx lang-dafnyx dafnyx colorized" data-line="1401" data-line-first="1402" style="display:block"><code data-line="1402"><span style="color:blue">lemma</span> IteratesAreFinite(x: A)
+  <span style="color:purple">requires</span> Acc(x) &amp;&amp; le(x, F(x))
+  <span style="color:purple">ensures</span> IsFinite(Iterates(x))
+{
+  IteratesAreAscending(x);
+  AscendingIsFinite(Iterates(x));
+}</code></pre>
+<p class="p noindent para-continued" data-line="1411"><span data-line="1411"></span>Note that this is an ordinary lemma (not an extreme lemma), since our proof
+simply uses other lemmas without any need to introduce the <span data-line="1412"></span><code class="code code1 language-dafnyx lang-dafnyx dafnyx colorized">_k</code><span data-line="1412"></span> index.
+</p><h3 id="sec-computing-the-least-fixpoint" class="h2" data-line="1414" data-heading-depth="2" style="display:block"><span data-line="1414"></span><span class="heading-before"><span class="heading-label">3.4</span>.&#8194;</span><span data-line="1414"></span>Computing the Least Fixpoint</h3>
+<p class="p noindent" data-line="1416"><span data-line="1416"></span>We<span data-line="1416"></span>&#39;<span data-line="1416"></span>ve finally arrived at where we can write a function that computes the
+least fixpoint of <span data-line="1417"></span><code class="code code1 language-dafnyx lang-dafnyx dafnyx colorized">F</code><span data-line="1417"></span>. We<span data-line="1417"></span>&#39;<span data-line="1417"></span>ll do it in two different way, first via our
+<span data-line="1418"></span><code class="code code1 language-dafnyx lang-dafnyx dafnyx colorized">Iterates</code><span data-line="1418"></span> function:
+</p>
+<pre class="para-block pre-fenced pre-fenced3 language-dafnyx lang-dafnyx dafnyx colorized" data-line="1420" data-line-first="1421" style="display:block"><code data-line="1421"><span style="color:blue">function</span> LFP(): A {
+  AboutIteratesOfBot();
+  Last(Iterates(bot))
+}</code></pre>
+<p class="p noindent para-continued" data-line="1427"><span data-line="1427"></span>The compiled function body is just <span data-line="1427"></span><code class="code code1 language-dafnyx lang-dafnyx dafnyx colorized">Last(Iterates(bot))</code><span data-line="1427"></span>. However, to use
+function <span data-line="1428"></span><code class="code code1 language-dafnyx lang-dafnyx dafnyx colorized">Last</code><span data-line="1428"></span>, we must meet its precondition that the given stream is finite.
+We gather that information by calling a lemma, which we state and prove thus:
+</p>
+<pre class="para-block pre-fenced pre-fenced3 language-dafnyx lang-dafnyx dafnyx colorized" data-line="1431" data-line-first="1432" style="display:block"><code data-line="1432"><span style="color:blue">lemma</span> AboutIteratesOfBot()
+  <span style="color:purple">ensures</span> IsFinite(Iterates(bot))
```

Comment:
Please look at the README on how you can ensure this blog post can be maintained with future versions of Dafny, especially for the `make check` commands